Manual/ModuleSystem.lean

```diff
@@ -0,0 +1,132 @@
+/-
+Copyright (c) 2025 Lean FRO LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Sebastian Ullrich
+-/
+
+import VersoManual
+import Manual.Meta
+
+open Verso.Genre Manual
+open Verso.Genre.Manual.InlineLean
+
+#doc (Manual) "The Module System" =>
+%%%
+%%%
+
+The module system is an experimental Lean 4 feature allowing for finer-grained control over what information is exported from Lean modules and imported into other modules, improving average build times, executable sizes, API design options, and more.
+
+The module system is activated by prefixing a Lean file with the `module` keyword.
+At the time of writing, it is additionally guarded by the `experimenta
+.module` option that has to be set to `true` in the project's Lakefile.
+`module`s can only import other `module`s so adoption has to be done top-down.
+Non-`module`s can import `module`s and will ignore all module system-specific annotations.
+
+# Visibility
+
+The main distinction the module system introduces is between the *public scope* that contains all information visible to other modules via `import` and the *private scope* that is not imported by default.
```

Comment:
```suggestion
The main distinction the module system introduces is between the {deftech}_public scope_ that contains all information visible to other modules via `import` and the {deftech}_private scope_ that is not imported by default.
```
This annotates them as technical terminology, which adds them to the quick jump box and makes them into link targets for use sites.